File: 2-algorithm/inter/romanNumConverter.js
/*jshint node: true */
'use strict';

function convertToRoman (num) {
  let result = '';
  let ones, tens, hunds, thous;
  let hundsA = ['C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM'];
  let tensA = ['X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC'];
  let onesA = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX'];
  thous = Math.floor(num / 1000);
  hunds = Math.floor(num % 1000 / 100);
  tens = Math.floor(num % 100 / 10);
  ones = Math.floor(num % 10);
  console.log(thous, hunds, tens, ones);
  for (let i = 0; i < thous;i++) {
    result += 'M';
  }
  if (hunds) {
    result += hundsA[hunds - 1];
  }
  if (tens) {
    result += tensA[tens - 1];
  }
  if (ones) {
    result += onesA[ones - 1];
  }
  return result;
}

console.log(convertToRoman(36));

File: old-beta/2-algorithm/interGo/romanNumConverter.go
package main

import (
	"fmt"
	"math"
)

func romanNumConverter() {
	var sol = make([]string, 0)
	for _, v := range test10 {
		sol = append(sol, getRomanNum(v.data))
	}
	for i, v := range sol {
		fmt.Println(i, v)
	}
}

func getRomanNum(num int) (sol string) {
	hundreds := []string{"C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"}
	tens := []string{"X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"}
	units := []string{"I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"}
	t := int(math.Floor(float64(num / 1000)))
	h := int(math.Floor(float64(num % 1000 / 100)))
	d := int(math.Floor(float64(num % 100 / 10)))
	u := int(math.Floor(float64(num % 10)))
	for i := 0; i < t; i++ {
		sol += "M"
	}
	if h > 0 {
		sol += hundreds[h-1]
	}
	if d > 0 {
		sol += tens[d-1]
	}
	if u > 0 {
		sol += units[u-1]
	}
	return sol
}

var test10 = []struct {
	data int
}{
	{36},
	{68},
	{697},
	{1000},
	{3999},
}
